modules/ngx_http_upstream_vnswrr_module/ngx_http_upstream_vnswrr_module.c

```diff
@@ -443,10 +482,10 @@ ngx_http_upstream_get_vnswrr(ngx_http_upstream_vnswrr_peer_data_t  *vnsp)
 
     if (peers->weighted) {
         /* batch initialization vpeers at runtime. */
-        if (uvnscf->vnumber != peers->total_weight
+        if (uvnscf->vnumber != peers->total_weight / peers->gcd
```

Comment:
Move field `gcd` into uvnscf. Then we'll not modify nginx core source round_robin.{c|h} .
(uvnscf->gcd). 

normal peers: uvnscf->gcd
backup peers: ubvnscf->gcd